internal/app/wwctl/container/rename/main_test.go

```diff
@@ -0,0 +1,63 @@
+package rename
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"path"
+	"testing"
+
+	containerList "github.com/hpcng/warewulf/internal/app/wwctl/container/list"
+	"github.com/hpcng/warewulf/internal/pkg/testenv"
+	"github.com/hpcng/warewulf/internal/pkg/warewulfd"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_Renname(t *testing.T) {
+	env := testenv.New(t)
+	env.WriteFile(t, path.Join(testenv.WWChrootdir, "test-container/rootfs/file"), `test`)
+	defer env.RemoveAll(t)
+	warewulfd.SetNoDaemon()
+
+	// first we will verify that there is an existing container
+	t.Run("container list", func(t *testing.T) {
+		verifyContainerListOutput(t, "test-container")
+	})
+
+	// then rename it
+	t.Run("container rename", func(t *testing.T) {
+		baseCmd := GetCommand()
+		baseCmd.SetOut(nil)
+		baseCmd.SetErr(nil)
+		baseCmd.SetArgs([]string{"test-container", "test-container-rename"})
+		err := baseCmd.Execute()
+		assert.NoError(t, err)
+	})
+
+	// retrieve again
+	t.Run("Container list", func(t *testing.T) {
+		verifyContainerListOutput(t, "test-container-rename")
+	})
+}
+
+func verifyContainerListOutput(t *testing.T, content string) {
+	baseCmd := containerList.GetCommand()
+	baseCmd.SetOut(nil)
+	baseCmd.SetErr(nil)
+	stdoutR, stdoutW, _ := os.Pipe()
```

Comment:
You could simply use `wwlog.SetLogWriter(buf)` instead of creatinga pipe and as stream

Reply:
> You could simply use `wwlog.SetLogWriter(buf)` instead of creatinga pipe and as stream

not sure if I understand correctly. Here we need to retrieve the output of `container list`, it is generated by https://github.com/JasonYangShadow/warewulf/blob/issue/583/internal/app/wwctl/container/list/main.go#L37, PrintHelper will directly output to os.Stdout.